lib/web/app/redirect.go

```diff
@@ -73,8 +77,13 @@ const js = `
           body: JSON.stringify(data),
         }).then(response => {
           if (response.ok) {
-            // redirect to the root and remove current URL from history (back button)
-            window.location.replace('/');
+            try {
+                window.location.replace(url.origin + path);
```

Comment:
I tried to test this myself locally but idk how to set up a custom webassets build and get all the redirects working.

Since anyone can share a link and modify the path parameter, we basically have to treat it as untrusted user input. Of course they could share any bad link they want, but if we're sending them to their secured teleport login page, I really want to make sure we're not going to redirect them anywhere weird.

Is there any way they could craft a path param that start with something like `.tld`? I'm imagining they share a link like `https://proxy.com/web/launch/app1.proxy.com%3Fpath=.io` so that someone would get redirected to `https://proxy.com.io` - a completely different website that could be in an attackers control

Reply:
Yeah its been a bit of a pain to get running/testing locally, I feel you there.

Very interesting to see that `.tld` vector, haven't personally seen that. Because any "real" path, including the default root path of "/" would all start with a forward slash, we can assert that any path given starts with that slash, otherwise redirect to origin only.